### PR TITLE
Add migration concept for old ctr app properties

### LINKED ISSUE #1715: Handle old ctr props
With new task launch ui we moved to not require task name in props anymore so that those are portable between tasks.
```
app.t1.timestamp.format=yyyy
```
On a server we still handle old style:
```
app.ctr1.t1.timestamp.format=yyyy
```
Need to think how we represent this in UI.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Add Some UI fu to let user to automatically migrate
  from old ctr app properties(given in 2.7.x line) and
  migrate those to new expected ctr options.
- For this work correctly also #1721 needs to be fixed which
  is a bigger topic and will get done in a separate commit.
- Fixes #1715

### CODE REVIEW

**Comment by cppwfs** [comment]
Create task ctrx with the following definition `t1:timestamp&&t2:timestamp`
1) Select launch for `ctrx` 
2) Set the timestamp format property  for each of the tasks (t1 & t2).
3) Launch the tasks
4) Immediately go back to launch the task `ctrx`  and you will see the properties are not set.
5) Refresh the page and the properties are there.

File: ui/src/app/tasks-jobs/tasks/launch/builder/builder.component.ts [hunk 1]
```diff
@@ -1022,4 +1030,53 @@ export class BuilderComponent implements OnInit, OnDestroy {
       this.launch.emit({props: this.getProperties(), args: this.getArguments()});
     }
   }
+
+  hasMigrations(): boolean {
+    return this.migrations && this.migrations.migratedMatch && this.migrations.migratedMatch.length > 0;
+  }
+
+  migrage() {
```
Comment (cppwfs) [inline ui/src/app/tasks-jobs/tasks/launch/builder/builder.component.ts:1038]: migrate?

Comment (jvalkeal) [reply ui/src/app/tasks-jobs/tasks/launch/builder/builder.component.ts:1038]: hah, funny how you don't see these typos.

Comment (cppwfs) [reply ui/src/app/tasks-jobs/tasks/launch/builder/builder.component.ts]: I hear ya'.   Done it more times that I can remember :-)

**Comment by cppwfs** [comment]
LGTM

**Comment by cppwfs** [comment]
Squashed Merged.

**Pull request closed without merging**